control-plane-operator/controllers/hostedcontrolplane/oauth/deployment_test.go

```diff
@@ -61,9 +68,19 @@ func TestReconcileOauthDeploymentNoChanges(t *testing.T) {
 		fakeClient := fake.NewClientBuilder().WithScheme(hyperapi.Scheme).Build()
 		oauthDeployment.Spec.MinReadySeconds = 60
 		expectedMinReadySeconds := oauthDeployment.Spec.MinReadySeconds
-		err := ReconcileDeployment(ctx, fakeClient, oauthDeployment, ownerRef, &tc.cm, imageName, tc.deploymentConfig, tc.serverParams.IdentityProviders(), tc.serverParams.OauthConfigOverrides,
+		err := ReconcileDeployment(ctx, fakeClient, oauthDeployment, tc.serverParams.AuditWebhookRef, ownerRef, &tc.cm, &tc.auditCM, imageName, tc.deploymentConfig, tc.serverParams.IdentityProviders(), tc.serverParams.OauthConfigOverrides,
 			tc.serverParams.AvailabilityProberImage, tc.serverParams.NamedCertificates(), tc.serverParams.Socks5ProxyImage, tc.serverParams.NoProxy, &tc.configParams, hyperv1.IBMCloudPlatform)
 		g.Expect(err).To(BeNil())
 		g.Expect(expectedMinReadySeconds).To(Equal(oauthDeployment.Spec.MinReadySeconds))
+		for _, container := range oauthDeployment.Spec.Template.Spec.Containers {
+			if container.Name == oauthContainerMain().Name {
```

Comment:
Is it a lot of work to move any of this to golden files? Test cases that do this are hard to reason about, getting a specific expectation on output per test case would be great.

Reply:
Do you mean to create a testdata directory to store the expected parameters? I don't find any existing testdata in the repo. I'm not sure if it's necessary to include such parameter tests in the e2e testing. I could also remove them if it's okay to ignore them. Thanks!

Reply:
We have a number of test fixtures - see uses of `CompareWithFixture`. This is a unit test, so it may be easy and quick to get those integrated instead of conditional validation logic. If you'd rather not do that, could you please refactor the unit test so that each test declares the validations it needs, instead of having a loosely-coupled conditional validation that applies to all in theory but only one in practice? Once you have expected-data-per-test, I think using `CompareWithFixture()` will be the fastest way to assert it.